Foundation/NSDecimal.swift

```diff
@@ -1,42 +1,159 @@
-    public static var greatestFiniteMagnitude: Decimal { NSUnimplemented() }
-    public static var leastNormalMagnitude: Decimal { NSUnimplemented() }
-    public static var leastNonzeroMagnitude: Decimal { NSUnimplemented() }
-    public static var pi: Decimal { NSUnimplemented() }
-    public var exponent: Int { NSUnimplemented() }
-    public var significand: Decimal { NSUnimplemented() }
-    public init(sign: FloatingPointSign, exponent: Int, significand: Decimal) { NSUnimplemented() }
-    public init(signOf: Decimal, magnitudeOf magnitude: Decimal) { NSUnimplemented() }
-    public var sign: FloatingPointSign { NSUnimplemented() }
-    public static var radix: Int { NSUnimplemented() }
-    public var ulp: Decimal { NSUnimplemented() }
+    public static let leastFiniteMagnitude = Decimal(
+        _exponent: 127,
+        _length: 8,
+        _isNegative: 1,
+        _isCompact: 1,
+        _reserved: 0,
+        _mantissa: (0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff)
+    )
+    public static let greatestFiniteMagnitude = Decimal(
+        _exponent: 127,
+        _length: 8,
+        _isNegative: 0,
+        _isCompact: 1,
+        _reserved: 0,
+        _mantissa: (0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff, 0xffff)
+    )
+    public static let leastNormalMagnitude = Decimal(
+        _exponent: -127,
+        _length: 1,
+        _isNegative: 0,
+        _isCompact: 1,
+        _reserved: 0,
+        _mantissa: (0x0001, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000)
+    )
+    public static let leastNonzeroMagnitude = Decimal(
+        _exponent: -127,
+        _length: 1,
+        _isNegative: 0,
+        _isCompact: 1,
+        _reserved: 0,
+        _mantissa: (0x0001, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000, 0x0000)
+    )
+    public static let pi = Decimal(
+        _exponent: -38,
+        _length: 8,
+        _isNegative: 0,
+        _isCompact: 1,
+        _reserved: 0,
+        _mantissa: (0x6623, 0x7d57, 0x16e7, 0xad0d, 0xaf52, 0x4641, 0xdfa7, 0xec58)
+    )
+    public var exponent: Int {
+        get {
+            return Int(self.__exponent)
+        }
+    }
+    public var significand: Decimal {
+        get {
+            return Decimal(_exponent: 0, _length: _length, _isNegative: _isNegative, _isCompact: _isCompact, _reserved: 0, _mantissa: _mantissa)
+        }
+    }
+    public init(sign: FloatingPointSign, exponent: Int, significand: Decimal) {
+        self.init(_exponent: Int32(exponent) + significand._exponent, _length: significand._length, _isNegative: sign == .plus ? 0 : 1, _isCompact: significand._isCompact, _reserved: 0, _mantissa: significand._mantissa)
+    }
+    public init(signOf: Decimal, magnitudeOf magnitude: Decimal) {         self.init(_exponent: magnitude._exponent, _length: magnitude._length, _isNegative: signOf._isNegative, _isCompact: magnitude._isCompact, _reserved: 0, _mantissa: magnitude._mantissa)
```

Comment:
Can you explain what you mean? Self.init is on line 143 and the public init is on line 142.